src/test/java/com/github/fppt/jedismock/comparisontests/scripting/EvalTest.java

```diff
@@ -294,4 +293,194 @@ public void evalUnpackTest(Jedis jedis) {
+    }
+
+    @TestTemplate
+    void evalCjsonEncodeJustArrayTest(Jedis jedis) {
+        assertThat(jedis.eval("return cjson.encode({1, nil, 3})", 0))
+                .isEqualTo("[1,null,3]");
+    }
+
+    @TestTemplate
+    void evalCjsonEncodeArrayWithMixedTypesTest(Jedis jedis) {
+        assertThat(jedis.eval("return cjson.encode({1, 'str', true, nil, {1, 2}, {['foo'] = 'bar'}})", 0))
+                .isEqualTo("[1,\"str\",true,null,[1,2],{\"foo\":\"bar\"}]");
+    }
+
+    @TestTemplate
+    void evalCjsonEncodeJsonBooleanTest(Jedis jedis) {
+        assertThat(jedis.eval("return cjson.encode({['foo'] = true})", 0))
+                .isEqualTo("{\"foo\":true}");
+    }
+
+    @TestTemplate
+    void evalCjsonEncodeJsonIntegerTest(Jedis jedis) {
+        assertThat(jedis.eval("return cjson.encode({['foo'] = 100})", 0))
+                .isEqualTo("{\"foo\":100}");
+    }
+
+    @TestTemplate
+    void evalCjsonEncodeJsonDoubleTest(Jedis jedis) {
+        assertThat(jedis.eval("return cjson.encode({['foo'] = 100.01})", 0))
+                .isEqualTo("{\"foo\":100.01}");
+    }
+
+    @TestTemplate
+    void evalCjsonEncodeJsonStringTest(Jedis jedis) {
+        assertThat(jedis.eval("return cjson.encode({['foo'] = 'str'})", 0))
+                .isEqualTo("{\"foo\":\"str\"}");
+    }
+
+    @TestTemplate
+    void evalCjsonEncodeNestedJsonTest(Jedis jedis) {
+        assertThat(jedis.eval("return cjson.encode({['foo'] = {['bar'] = 'baz'}})", 0))
+                .isEqualTo("{\"foo\":{\"bar\":\"baz\"}}");
+    }
+
+    @TestTemplate
+    void evalCjsonEncodeJsonArrayTest(Jedis jedis) {
+        assertThat(jedis.eval("return cjson.encode({['foo'] = {'bar', 'baz'}})", 0))
+                .isEqualTo("{\"foo\":[\"bar\",\"baz\"]}");
+    }
+
+    @TestTemplate
+    void evalCjsonEncodeJsonArray(Jedis jedis) {
+        assertThat(jedis.eval("return cjson.encode({{['foo'] = 'bar'}})", 0))
+                .isEqualTo("[{\"foo\":\"bar\"}]");
+    }
+
+    @TestTemplate
+    void evalCjsonEncodeNilTest(Jedis jedis) {
+        assertThat(jedis.eval("return cjson.encode(nil)", 0))
+                .isEqualTo("null");
+    }
+
+    @TestTemplate
+    void evalCjsonEncodeEmptyTableTest(Jedis jedis) {
+        assertThat(jedis.eval("return cjson.encode({})", 0))
+                .isEqualTo("{}");
+    }
+
+    @TestTemplate
+    void evalCjsonDecodeBooleanTest(Jedis jedis) {
+        assertThat(jedis.eval("return cjson.decode('true')", 0))
+                .isEqualTo(1L);
+    }
+
+    @TestTemplate
+    void evalCjsonDecodeIntegerTest(Jedis jedis) {
+        assertThat(jedis.eval("return cjson.decode('100')", 0))
+                .isEqualTo(100L);
+    }
+
+    @TestTemplate
+    void evalCjsonDecodeDoubleTest(Jedis jedis) {
+        assertThat(jedis.eval("return tostring(cjson.decode('100.01'))", 0))
+                .isEqualTo("100.01");
+    }
+
+    @TestTemplate
+    @Disabled("Failed to make dkjson return correct value for string")
```

Comment:
it will possibly require to make another one patch to dkjson sources to support such behaviour